services/navigation/path.go

```diff
@@ -32,16 +42,26 @@ func (p *Path) GeoPoints() []*geo.Point {
 }
 
 // PathSliceToProto converts a slice of Path into an equivalent Protobuf message.
-func PathSliceToProto(paths []*Path) []*pb.Path {
+func PathSliceToProto(paths []*Path) ([]*pb.Path, error) {
+	if paths == nil {
```

Comment:
do we care if it is nil or if it is empty? If its emptiness we care about then we should be checking length.

Reply:
it is safe to call range on a nil / empty slice, it just won't perform any iterations:
```
package main

import "fmt"

func main() {
	a := []int{}
	for _ = range a {
		fmt.Println("I don't execute")
	}
	a = nil
	for _ = range a {
		fmt.Println("I ALSO don't execute")
	}
	fmt.Println("done!")
}
```

that prints `done!`

Reply:
Since it's safe I don't think it is necessary to have a check at all, I will remove it. 
Thanks for pointing this out.